packages/SwingSet/test/upgrade/test-upgrade.js

```diff
@@ -0,0 +1,78 @@
+// eslint-disable-next-line import/order
+import { test } from '../../tools/prepare-test-env-ava.js';
+
+// eslint-disable-next-line import/order
+import { assert } from '@agoric/assert';
+import { parse } from '@endo/marshal';
+import { provideHostStorage } from '../../src/controller/hostStorage.js';
+import { initializeSwingset, makeSwingsetController } from '../../src/index.js';
+import { capargs } from '../util.js';
+
+function bfile(name) {
+  return new URL(name, import.meta.url).pathname;
+}
+
+async function testUpgrade(t, defaultManagerType) {
+  const config = {
+    defaultManagerType,
+    bootstrap: 'bootstrap',
+    defaultReapInterval: 'never',
+    vats: {
+      bootstrap: { sourceSpec: bfile('bootstrap-upgrade.js') },
+    },
+    bundles: {
+      ulrik1: { sourceSpec: bfile('vat-ulrik-1.js') },
+      ulrik2: { sourceSpec: bfile('vat-ulrik-2.js') },
```

Comment:
"ulrik"?  Inquiring minds want to know...

Reply:
Needed a name that started with U, for Ulrik the Upgraded (vat), like Tammy the Terminated or Pauline the Pipelined (neither of which are examples we use but we totally should). And all the other "U" baby names I found online were in norwegian or icelandic and contained characters I don't know how to type :)

Reply:
"Ulrik the Upgraded" sound like one of the less well known kings of Sweden.